### PR TITLE
LKE-10595 fix(CustomIcons): pass baseUrl to ogma parser

### PR DESCRIPTION
**Pull request by aminerhanemi**
To fix the [issue](https://linkurious.atlassian.net/browse/LKE-10595) we had to use a relative path when storing custom icon URLs in the style object.
 That creates [the issue ](https://linkurious.atlassian.net/browse/LKE-6952) where custom icons are no longer visible in the export plugin.
 
A possible fix is to add a base element to the index file.
It smees working but I am not sure if it is done the right way.

 and there is still an issue as the icon is not visible in the preview in the bottom right corner
 
Screenshot 2024-03-06 at 15 34 48

**Solution**
Now we pass the baseUrl to Ogma parser.

### CODE REVIEW

**Comment by w8r** [comment]
And does it work?
I would put it into `src/main.tsx` for the lack of a better option

**Comment by aminerhanemi** [comment]
> And does it work? I would put it into `src/main.tsx` for the lack of a better option

It works in the viz and the export but not in the bottom right corner preview

**Comment by w8r** [comment]
> It works in the viz and the export but not in the bottom right corner preview

That's alright, I think images are suppressed there

File: src/api.ts [hunk 1]
```diff
@@ -9,7 +9,8 @@ import {
 declare let IS_DEV: boolean;
 
 const rc = new RestClient({
-  baseUrl: IS_DEV ? "http://localhost:3000/" : "../../",
+  // we take the first part of the url to get the base url (needed in case using baseFolder)
+  baseUrl: IS_DEV ? "http://localhost:3000/" : document.location.href.split('plugins')[0],
```
Comment (aminerhanemi) [inline src/api.ts:13]: change needed as now we added a base element

**Comment by aminerhanemi** [comment]
@w8r e2e tests fail because of ogma version update. I think there is a need to update the screenshot. I don't know how you generate the image so I am not able to fix it, maybe you can show me how.

**Review by davidrapin — APPROVED** [review]
Ok for me :)

**Comment by aminerhanemi** [comment]
> npm run replace-refs

still failing :/

**Pull request merged**